src/wasm/wasm-ir-builder.cpp

```diff
@@ -2214,15 +2214,17 @@ Result<> IRBuilder::makeBrOn(
 Result<> IRBuilder::makeStructNew(HeapType type) {
   StructNew curr(wasm.allocator);
   curr.type = Type(type, NonNullable, Exact);
-  // Differentiate from struct.new_default with a non-empty expression list.
   curr.operands.resize(type.getStruct().fields.size());
-  CHECK_ERR(visitStructNew(&curr));
-  push(builder.makeStructNew(type, std::move(curr.operands)));
+  CHECK_ERR(ChildPopper{*this}.visitStructNew(&curr));
```

Comment:
Since the ChildPopper visitor doesn't take any additional arguments, this can remain simply `visitStructNew` here and below.